app/javascript/components/widgets/numbers/fallingblocks.jsx

```diff
@@ -0,0 +1,62 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+/* global UnityLoader */
+
+function shuffleArray(oldArray) {
+  const array = oldArray;
+  for (let i = array.length - 1; i > 0; i -= 1) {
+    const j = Math.floor(Math.random() * (i + 1));
+    [array[i], array[j]] = [array[j], array[i]];
+  }
+}
+
+class FallingBlocks extends React.Component {
+  constructor(props) {
+    super(props);
+    this.blocks = [];
```

Comment:
I think we could skip the array creation and shuffle by keeping pr, commit, slack counts in state and randomly decrementing any of the fields that is greater than zero.

Reply:
I like it, much simpler. 

This ends up changing the distribution of the block types so each group is weighted evenly instead of having them weighted by the number of blocks of each type. If one group is much larger then the others, once the other types have been depleted you get a bunch of that type all at once.

Reply:
You can maintain standard distribution by selecting a random integer from the range of 0..sum(counts) and selecting the decrement index based on the ranges.

Reply:
I'll stick with the simpler way for now, if we decide its important we can change it later